### ISSUE TITLE
Raw values

### ISSUE THREAD

**Issue by steeph-k** [kitschpatrol/Brain#7]
Hi!

Has any work already been done regarding raw EEG values from the ThinkGear module? Switching the mode works, so I do get values that look like ThinkGear packets at 57.6k. But the routine provided by the library is not able to parse them. I'm not sure where to start to adapt it to work with the different packages. I'm still trying to understand how the code exactly does what it does.

If anybody could give me a hint, that would be awesome!
Thanks!

**Comment by zombodotcom** [comment]
Hey, I got a mindflex duel, so 2 headsets, I'm only testing one headset so far, I turn the headset on and I can get data for a few seconds on the braingrapher with an arduino, then I'm getting 200,0,0 some other guy had the same question and I asked him on twitter with one question and he blocked me for some reason? 
I get a few seconds of data every time I turn it on and off again then it cuts out to 200,0,0 

Any ideas ? I'll try the other headset when I have time

**Comment by steeph-k** [comment]
How is the headset connected? Is it using a common ground with the Arduino? Just an idea.

**Comment by sicklebrick** [comment]
Too much noise that way, use an optoisolator/optocoupler to connect the
mindflex to the arduino and you'll get an awesome signal.

**Comment by blipi** [comment]
I am using common ground, but of course as @sicklebrick says an optoisolator between both would be the ideal way to avoid having noise.

**Comment by zombodotcom** [comment]
Yes it is on the ground pin of the arduino , the funny thing is that when I unpower the arduino it still has lights blinking while is connected to rx and gnd !

**Comment by zombodotcom** [comment]
Anyone know why I'm gettinga few seconds of data then it cuts out to 200, 0,0 using rx pin and gnd

**Comment by Riteshlogin** [comment]
@zombodotcom I know this is a bit late, but it will only give you 200 if the TGAM feels that the electrode is not contacting any skin.

I have my own question. So I managed to get my hands on some standalone TGAM modules that were already hardware-configured at the 57.6k baud mode. I am giving the TGAM a proper amount of power, I have ear clips attached as reference electrodes and another electrode for my forehead.  I then uploaded the sample code to an Arduino and started reading the data. 

However, though I am constantly getting meditation and attention values, I never seem to be able to get alpha, beta, and gamma values. They only really come once in a blue moon. Has anyone else had this problem before?

**Comment by gpascualg** [comment]
@zombodotcom As far as I remember (not much really...) there are 2 ways of setting the mode (aside of the baudrate):

1. By soldering a contact on the board, in which case you will have to research which and unsolder/solder it.

2. If not manually overriden by 1), you can tell the TGAM module to give you the values, see:
* TGAM 1.6: https://github.com/gpascualg/TGAM1-EEGReader/blob/master/EEGReader/TGAM16.cpp#L29 and https://github.com/gpascualg/TGAM1-EEGReader/blob/master/EEGReader/TGAM16.cpp#L46
* TGAM 1.7: https://github.com/gpascualg/TGAM1-EEGReader/blob/master/EEGReader/TGAM17.cpp#L18 and https://github.com/gpascualg/TGAM1-EEGReader/blob/master/EEGReader/TGAM17.cpp#L38

Basically, it has the RAW output disabled, which are those alpha/beta/gamma values

**Comment by Riteshlogin** [comment]
But I do receive alpha, beta, gamma, etc. values. But only very rarely, whereas I have read that it is typical to get them at least once every second.

**Comment by gpascualg** [comment]
Oh, I though you did not receive them at all. That's strange, it might be related to the signal strength, have you checked it? I believe it will only send those values if it is confident the signal received is strong enough.

On #12 some methods to improve the signal are discussed, try them too just in case.

**Comment by Riteshlogin** [comment]
And is a signal quality value of 48-52 normal? That's what its been fluctuating around for me.

I'll let you know if anything on that thread you posted works.

**Comment by steeph-k** [comment]
I've linked the spec sheet with descriptions on how to change the mode in the beginning of this issue: https://github.com/kitschpatrol/Brain/issues/7#issuecomment-37422069

Alpha, beta, gamma etc. values are not raw data. Raw output means actual raw values, not FFTed values.

**Comment by Riteshlogin** [comment]
Again, I actually do manage to receive alpha, beta, and gamma values. It's not that I cant receive them, it's that they don't come as nearly as often as the documentation says they should (about once per second).

**Comment by steeph-k** [comment]
I understand. Sorry, I've never had this problem. My comment was more meant for gpascualg because it sounded like he confused those with raw data.

**Comment by gpascualg** [comment]
@steeph-k True, it's been a while since I last played with the TGAM =P. Still, the links on https://github.com/kitschpatrol/Brain/issues/7#issuecomment-228798284 also contain code to change to FFT mode instead of RAW.

**Comment by gpascualg** [comment]
@Riteshlogin According to http://developer.neurosky.com/docs/doku.php?id=helloeeg_adobe_flash_tutorial signal strength goes from 0 (perfect) to 200 (bad). 

You should, theorically, be reading 0. Any other value should indicate the signal is not as strong as required: http://www.darrenmothersele.com/blog/2013/09/21/brain-wave-visualisation/

That said, you can try using opto-isolators, grounding it somewhere else (I, for example, did not ground it to the arduino itself but to the battery pack, so both the arduino and TGAM where directly grounded to the battery).

**Comment by Riteshlogin** [comment]
@steeph-k My apologies for misunderstanding you!

**Comment by Riteshlogin** [comment]
Welp, I just gave the TGAM datasheet another read and discovered that I had been speed reading over the part where it says that shielded wire is necessary for the electrodes. I'm fairly certain now that that was my problem.

**Comment by penco193** [comment]
Hi!, i have a mindflex game with ThinkGear module rev 2.7. any of you can tell me if the firmware version is 1.6 or 1.7 for this board?, or where could i find it?. any help will be appreciated

**Comment by TimoteoMendes** [comment]
Hello,

I am trying to use a simple tester just to see the raw data, but most of the time the values have error in crc or long.
The @gpascualg has way to much code for what i need.
Is it possible to find out the firmware version?
I have the rev2.9 board.
Thanks

**Comment by kitschpatrol** [comment]
Thanks for the conversation. Closing this as it's quite old.
